source/ListingCommands.cpp

```diff
@@ -0,0 +1,333 @@
+  
+  std::cout << "\nBuilt-in filters:" << std::endl;
+  std::cout << "  • source-filter: Filters sources based on criteria" << std::endl;
+  std::cout << "  • sink-filter: Filters sinks based on criteria" << std::endl;
+  std::cout << "  • transform-filter: Filters transform operations" << std::endl;
+  std::cout << "  • method-filter: Filters methods from analysis" << std::endl;
+  std::cout << "  • class-filter: Filters classes from analysis" << std::endl;
+  
+  std::cout << "\nNote: Custom filters can be defined in configuration files." << std::endl;
+}
+
+void ListingCommands::list_all_features(Context& /* context */) {
+  std::cout << "=== All Features ===" << std::endl;
+  
+  std::cout << "\nBuilt-in features:" << std::endl;
+  std::cout << "  • always-type: Always includes type information in traces" << std::endl;
+  std::cout << "  • via-type-of: Adds type information of specific arguments" << std::endl;
+  std::cout << "  • via-value-of: Adds value information of specific arguments" << std::endl;
+  std::cout << "  • via-cast: Indicates data type casting operations" << std::endl;
+  std::cout << "  • first-field: Marks first field access in a chain" << std::endl;
+  std::cout << "  • first-index: Marks first array/list index access" << std::endl;
+  std::cout << "  • broadening: Indicates where path broadening occurred" << std::endl;
+  std::cout << "  • return-via-parameter: Data returned through parameter modification" << std::endl;
+  std::cout << "  • intent-routing: Android Intent routing patterns" << std::endl;
+  std::cout << "  • copy-arguments: Argument copying behavior" << std::endl;
+  std::cout << "  • origin-field: Original field where taint was introduced" << std::endl;
+  std::cout << "  • array-allocation: Array allocation tracking" << std::endl;
+  std::cout << "  • breadcrumbs: Execution path tracking" << std::endl;
+  
+  std::cout << "\nCustom features can be defined in model generators and configuration files." << std::endl;
+}
+
+void ListingCommands::list_all_lifecycles(Context& context) {
+  std::cout << "=== All Lifecycles ===" << std::endl;
+  
+  auto lifecycles_paths = context.options->lifecycles_paths();
+  
+  for (const auto& lifecycle_path : lifecycles_paths) {
+    std::cout << "\nLifecycles from: " << lifecycle_path << std::endl;
+    
+    try {
+      Json::Value lifecycle_json = JsonReader::parse_json_file(lifecycle_path);
+      
+      for (const auto& lifecycle : lifecycle_json) {
+        if (lifecycle.isMember("name")) {
+          std::cout << fmt::format("  Name: {}", lifecycle["name"].asString()) << std::endl;
+        }
+        if (lifecycle.isMember("methods")) {
+          std::cout << "  Methods:" << std::endl;
+          for (const auto& method : lifecycle["methods"]) {
+            if (method.isString()) {
+              std::cout << fmt::format("    - {}", method.asString()) << std::endl;
+            }
+          }
+        }
+        std::cout << std::endl;
```

Comment:
Can we use `LifecycleMethod::from_json` instead of re-parsing attributes manually?